types.cc

```diff
@@ -1822,6 +1822,29 @@ static void serialize(const abstract_type& t, const void* value, bytes::iterator
     return ::serialize(t, value, out, cql_serialization_format::internal());
 }
 
+template <FragmentedView View>
+data_value collection_type_impl::deserialize(View v, cql_serialization_format sf) const {
+    struct visitor {
+        View v;
+        cql_serialization_format sf;
+        data_value operator()(const abstract_type&) { abort(); }
```

Comment:
Use on_internal_error() so as not to upgrade a but to total loss of availability.

Reply:
I will do that, but this code is *very* unreachable (a new collection type would need to be added) so it doesn't really matter.